### DISCUSSION TITLE
V2中如何与antd 3.x共存

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#1281] [Q&A 2.x] [answered]
场景：
1.现有项目依赖的是antd 3.x 
2.基于formily v2发布了一个npm包 package1。package1被项目所依赖。

问题：
安装依赖时，formily被扁平化安装到了项目根目录node_modules；
antd 4.x被安装到了package1中的node_modules；

最终导致报错，如下图：
![image](https://user-images.githubusercontent.com/18549980/115814463-af7fcc00-a427-11eb-91de-ce75fb432e55.png)

希望结果：
formily v2被安装到package1中。奈何没找到npm相关的配置方式。

**Comment by janryWang** [comment, marked as answer]
这个是因为formily/antd是走的peerDependencies，内部不会默认安装antd，所以是无法共存的，为的就是解决重复打包的问题

**Reply by NexxLuo** [reply]
有没有其它方式能解决这个问题？比如通过npm中的啥配置，强制像npm2中的树形安装依赖？很苦恼啊。。。

**Reply by NexxLuo** [reply]
我直接把@formily/antd给复制一份到自己项目把，或者把npm包给封装一层作为项目的依赖
